flink-core/src/main/java/org/apache/flink/api/common/typeutils/CompatibilityResult.java

```diff
@@ -60,10 +62,10 @@
 	 *
 	 * @return a result that signals migration is necessary, also providing a convert deserializer.
 	 */
-	public static <T> CompatibilityResult<T> requiresMigration(TypeDeserializer<T> convertDeserializer) {
+	public static <T> CompatibilityResult<T> requiresMigration(@Nonnull TypeDeserializer<T> convertDeserializer) {
 		Preconditions.checkNotNull(convertDeserializer, "Convert deserializer cannot be null.");
 
-		return new CompatibilityResult<>(true, Preconditions.checkNotNull(convertDeserializer));
+		return new CompatibilityResult<>(true, convertDeserializer);
```

Comment:
I've removed this `Preconditions.checkNotNull` because it was already done in the factory method (see L66). So this one is just redundant, the removal is not related to the `Nonnull` annotation.